crates/libs/wallet/src/types/compact_wad.rs

```diff
@@ -12,12 +12,20 @@ use nuts::traits::Unit;
 
 use serde::de::DeserializeOwned;
 use serde::{Deserialize, Serialize};
+use tonic::Status;
 
 use super::NodeUrl;
 
 use bitcoin::base64::engine::{GeneralPurpose, general_purpose};
 use bitcoin::base64::{Engine as _, alphabet};
 
+#[derive(Debug, Clone)]
+pub struct DleqProofError {
+    pub proof_index: usize,
+    pub amount: Amount,
```

Comment:
Is the amount really needed? It can be retrieved from the original vec + proof_index